### PR TITLE
Aliu/fd2 multichip

### PR DESCRIPTION
**Pull request by aliuTT**

### CODE REVIEW

**Review by ubcheema — APPROVED** [review]

File: tt_metal/impl/device/device.cpp [hunk 1]
```diff
@@ -1608,9 +1608,9 @@ bool Device::close() {
         for (auto it = not_done_dispatch_cores.begin(); it != not_done_dispatch_cores.end(); it++) {
             const auto &phys_core = *it;
             if(llrt::is_ethernet_core(phys_core, this->id_)) {
-                log_info(tt::LogMetal, "Ethernet dispatch core {} on Device {} is idle. Closing Device {}", phys_core.str(), mmio_device_id, this->id());
+                log_debug(tt::LogMetal, "Ethernet dispatch core {} on Device {} is idle. Closing Device {}", phys_core.str(), mmio_device_id, this->id());
             } else {
-                log_info(tt::LogMetal, "Resetting core {} on Device {} when closing Device {}", phys_core.str(), mmio_device_id, this->id());
+                log_debug(tt::LogMetal, "Resetting core {} on Device {} when closing Device {}", phys_core.str(), mmio_device_id, this->id());
```
Comment (pgkeller) [inline tt_metal/impl/device/device.cpp:1613]: ok for this PR, but in general it is good to keep these cleanups in separate commits in case regressions are introduced

File: tt_metal/impl/device/multi_device.hpp [hunk 2]
```diff
@@ -21,10 +21,11 @@ class DeviceMesh
 {
 public:
     DeviceGrid device_grid;
-    std::vector<std::pair<int, std::unique_ptr<Device>>> managed_devices;
+    std::map<chip_id_t, Device *> managed_devices;
```
Comment (pgkeller) [inline tt_metal/impl/device/multi_device.hpp:24]: chip_id_t is an int, right?  use a vector?

Comment (aliuTT) [reply tt_metal/impl/device/multi_device.hpp:24]: for a lot of use cases we create device 0 and device 7, just because of how they are connected in a T3000. Can still be a vector in that case, but I'm cleaning this up as part of the DevicePool changes, so I'll leave this alone for now so multichip CI can pass..

File: tt_metal/impl/device/multi_device.cpp [hunk 3]
```diff
@@ -53,17 +62,29 @@ std::vector<Device*> DeviceMesh::get_devices() const
...
 
 int DeviceMesh::num_devices() const
 {
-    return managed_devices.size();
+    return mesh_devices.size();
 }
 
+void DeviceMesh::close_devices() {
+    // TODO: change api to a yield, shouldn't allow closing sub grids in a pool of devices
+    tt::Cluster::instance().set_internal_routing_info_for_ethernet_cores(false);
```
Comment (cfjchu) [inline tt_metal/impl/device/multi_device.cpp:78]: Are there any issues if those `tt::Cluster::instance().set_internal_routing_info_for_ethernet_cores(false);` is called twice (i.e. explicit close_devices() and destructor)? Or should we guard this by checking if `not managed_devices.empty()`

**Review by cfjchu — APPROVED** [review]

**Review by mo-tenstorrent — APPROVED** [review]

File: tt_metal/impl/device/device.cpp (hunk 1 above)
Comment (aliuTT) [reply tt_metal/impl/device/device.cpp:1613]: I'll make two commits. I think there's non-zero chance there are some regressions 🙃

**Review by pgkeller — APPROVED** [review]
